awsbestpractices/s3-with-prefix/S3WithPrefix.java

```diff
@@ -0,0 +1,69 @@
+package example;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3URI;
+import com.amazonaws.services.s3.model.*;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+public class Example {
+
+    private final AmazonS3 s3Client;
+
+    public static double conditionWithPagination1(List<String> inputURIs) {
+        Long numBytes = 0L;
+        for (String uri : inputURIs) {
+            AmazonS3URI s3URI = new AmazonS3URI(uri);
+            ObjectListing objects = null;
+            do {
+                if (objects == null) {
+                    objects = s3Client.listObjects(s3URI.getBucket(), s3URI.getKey());
+                } else {
+                    //Should be flagged
+                    objects = s3Client.listNextBatchOfObjects(objects);
+                }
+                for (S3ObjectSummary summary : objects.getObjectSummaries()) {
+                    numBytes += summary.getSize();
+                }
+            } while (objects.isTruncated());
+        }
+
+        return null;
+    }
+
+    public void conditionWithPagination2(){
+        ListObjectsV2Request req = new ListObjectsV2Request().withBucketName(bucketName).withMaxKeys(2);
+        ListObjectsV2Result result;
+
+        do {
+            //Should be flagged
+            result = s3Client.listObjectsV2(req);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is manually paginating results. You could use the SDK to automatically paginate these results instead. SDK operation name: [S3Objects.withPrefix](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/iterable/S3Objects.html#withPrefix-com.amazonaws.services.s3.AmazonS3-java.lang.String-java.lang.String-).
 The list of [S3ObjectSummary](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/S3ObjectSummary.html)s will be fetched lazily, a page at a time, as they are needed. The size of the page can be controlled with the [withBatchSize(int)](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/iterable/S3Objects.html#withBatchSize-int-) method.